lib/internal/modules/cjs/loader.js

```diff
@@ -297,8 +327,58 @@ function findLongestRegisteredExtension(filename) {
   return '.js';
 }
 
+// This only applies to requests of a specific form:
+// 1. name/.*
+// 2. @scope/name/.*
+const EXPORTS_PATTERN = /^((?:@[^./@\\][^/@\\]*\/)?[^@./\\][^/\\]*)(\/.*)$/;
+function resolveExports(nmPath, request, absoluteRequest) {
+  // The implementation's behavior is meant to mirror resolution in ESM.
+  if (experimentalExports && !absoluteRequest) {
+    const [, name, expansion] =
+      request.match(EXPORTS_PATTERN) || [];
```

Comment:
get `StringPrototype` off `primordials` and then use `StringPrototype.match(request, EXPORTS_PATTERN)`. This avoid mutation of `String.prototype` from affecting this code.